src/lib_ccx/ccx_encoders_common.c

```diff
@@ -226,10 +217,7 @@ int change_utf8_encoding(unsigned char* dest, unsigned char* src, int len, enum
 						| (((unsigned int)(src[2] & 0x3F)) << 12U)
 						| (((unsigned int)(src[3] & 0x3F)) << 6U)
 						|  ((unsigned int)(src[4] & 0x3FU)) );
-					if (c < 256)
-						*(dest++) = c;
-					else
-						*dest++ = '?';
+					*(dest++) = c;
```

Comment:
This is not correct.
I understand that of course c is always going to be less than 256 because it's a char here, but if look at the function where it takes its value it returns an integer - so what we need to change is the type of c.